enzyme/Enzyme/CacheUtility.cpp

```diff
@@ -459,14 +459,18 @@ llvm::AllocaInst *CacheUtility::getDynamicLoopLimit(llvm::Loop *L,
     auto Limit = B.CreatePHI(found.var->getType(), 1);
 
     for (BasicBlock *Pred : predecessors(ExitBlock)) {
-      if (LI.getLoopFor(Pred) == L) {
+      if (LI.getLoopFor(Pred) == L || L->contains(Pred)) {
```

Comment:
Okay I agree with this change, as previously this had an assumption that an ExitBlcok would only be the exitblock for a single loop.

Will try to ponder a different test of this regarding the other changes.